requirements-test.txt

```diff
@@ -0,0 +1,4 @@
+six
+backports.tempfile
+nose
+mock
```

Comment:
This is not something needed for the new test, right? It is just something missing. If so, would be great to have it as a separate commit.

Then, I suggest to aggregate all the trivial patches that are sort of independent of this ZFS check into a separate pull request (this one, the .gitignore change, the first commit which cleans up the test). Then we could go ahead and merge those quickly and independently.